libfwupdplugin/fu-self-test.c

```diff
@@ -1390,6 +1390,36 @@ fu_device_inhibit_func (void)
 	g_assert_false (fu_device_has_flag (device, FWUPD_DEVICE_FLAG_UPDATABLE_HIDDEN));
 }
 
+#define TEST_FLAG_FOO	(1 << 0)
+#define TEST_FLAG_BAR	(1 << 1)
+
+static void
+fu_device_private_flags_func (void)
+{
+	g_autofree gchar *tmp = NULL;
+	g_autoptr(FuDevice) device = fu_device_new ();
+
+	fu_device_register_private_flag (device, TEST_FLAG_FOO, "foo");
+	fu_device_register_private_flag (device, TEST_FLAG_BAR, "bar");
+
+	fu_device_set_custom_flags (device, "foo");
+	g_assert_cmpint (fu_device_get_private_flags (device), ==, TEST_FLAG_FOO);
+	fu_device_set_custom_flags (device, "bar");
+	g_assert_cmpint (fu_device_get_private_flags (device), ==, TEST_FLAG_FOO | TEST_FLAG_BAR);
+	fu_device_set_custom_flags (device, "~bar");
+	g_assert_cmpint (fu_device_get_private_flags (device), ==, TEST_FLAG_FOO);
+	fu_device_set_custom_flags (device, "baz");
+	g_assert_cmpint (fu_device_get_private_flags (device), ==, TEST_FLAG_FOO);
+
```

Comment:
I think you want another test case of trying to use an unregistered flag `baz`.

Reply:
like the above? Or did you expect the baz value to appear in the to_string output, e.g. `PrivateFlags:         foo,unknown:4\n` etc?

Reply:
I think there should be some error here somehow - you shouldn't be able to add a private flag when you haven't registered it.  Maybe actually it should be an assertion.

Reply:
I went with critical warning, as then you can test for it in the self tests.